src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,189 +1,376 @@
-            val isMasterMode = dungeon.contains("Master Mode")
-
-            event.stackTip = if (floor.contains("Entrance")) {
-                "E"
-            } else if (isMasterMode) {
-                "M${floorNum}"
-            } else {
-                "F${floorNum}"
-            }
+            stackTip(event)
+
+            highlightingHandler(event)
+
+            toolTipHandler(event)
         }
-    }
 
-    @SubscribeEvent
-    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
-        if (!LorenzUtils.inSkyBlock || LorenzUtils.skyBlockArea != "Dungeon Hub") return
-        if (event.inventoryName != "Catacombs Gate") return
+    }
 
-        val lore = event.inventoryItems[45]?.getLore() ?: return
+    private fun stackTip(event: InventoryFullyOpenedEvent) {
+        val inventoryName = event.inventoryName
+        if (catacombsGatePattern.matches(inventoryName)) catacombsGateStackTip(event)
+        if (!config.floorAsStackSize) return
+        if (selectFloorPattern.matches(inventoryName)) selectFloorStackTip(event)
+        if (partyFinderTitlePattern.matches(inventoryName)) partyFinderStackTip(event)
+    }
 
-        if (lore[0] == "§7View and select a dungeon class.") {
-            selectedClass = lore[2].split(" ").last().removeColor()
+    private fun selectFloorStackTip(event: InventoryFullyOpenedEvent) {
+        inInventory = true
+        for ((slot, stack) in event.inventoryItems) {
+            val name = stack.displayName.removeColor()
+            if (anyFloorPattern.matches(name)) {
+                floorStackSize[slot] = "A"
+            } else if (entranceFloorPattern.matches(name)) {
+                floorStackSize[slot] = "E"
+            } else if (floorPattern.matches(name)) {
+                floorStackSize[slot] =
+                    floorNumberPattern.matchMatcher(name) { group("floorNum").romanToDecimalIfNecessary().toString() }
+                        .toString()
+            }
         }
     }
 
-    @SubscribeEvent
-    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
-        if (!LorenzUtils.inSkyBlock) return
-        if (event.gui !is GuiChest) return
+    private fun partyFinderStackTip(event: InventoryFullyOpenedEvent) {
+        inInventory = true
+        for ((slot, stack) in event.inventoryItems) {
+            val name = stack.displayName.removeColor()
+            if (checkIfPartyPattern.matches(name)) {
+                val lore = stack.getLore()
+                val floor = lore.find { floorFloorPattern.matches(it.removeColor()) }
+                val dungeon =
+                    lore.find { dungeonFloorPattern.matches(it.removeColor()) }
+                if (floor == null || dungeon == null) continue
+                val floorNum =
+                    floorNumberPattern.matchMatcher(floor) { group("floorNum").romanToDecimalIfNecessary().toString() }
+                        .toString()
```

Comment:
here this second .toString can either be replaced with ?: continue or just removed